### ISSUE TITLE
Upgrading from 1.3.0 to 1.4.2 does not work

### ISSUE THREAD

**Issue by mkobit** [topicusfinan/bitbucket-webhooks-plugin#41]
We have tried upgrading from 1.3.0 to 1.4.2 and all previous functionality stops working.

### What we tried:

1. Push up a new branch to a project and immediately saw it get indexed
1. Open up a pull request, immediately see it indexed and triggered
1. Upgrade this plugin on Bitbucket Server from **1.3.0** to **1.4.2**
1. Push up a new branch - no new job indexed or triggered
1. Open up a PR - no new job indexed or triggered
1. Restart Bitbucket Server
1. Repeat the branch and PR steps, get same results
1. Roll back plugin to **1.3.0**
1. Repeat branch and PR steps, expected jobs created and triggered

We tried looking at logs (did not have debug mode enabled) on both Jenkins and Bitbucket, and saw nothing.

### Versions

Bitbucket Server version: **4.13.0**
Bitbucket Branch Source: **2.1.2**
Jenkins: **2.55**

**Comment by casz** [comment]
Tried rebooting bitbucket after upgrading plugin?
Also jenkins?
Cause we had the same issue but a simple reboot fixed it.

**Comment by mkobit** [comment]
@casz did you also have to reboot Jenkins?

**Comment by casz** [comment]
Yes to ensure no lasting webhook was connected and sending request 👍

**Comment by mkobit** [comment]
@casz Thanks, I'll give it a try and report back!

**Comment by casz** [comment]
You could also try removing the plugin and installing it again, cause Jenkins should auto register when jenkins decides to scan

**Comment by casz** [comment]
I went as far as creating a jira issue on jenkins: https://issues.jenkins-ci.org/browse/JENKINS-43541 before trying a reboot of both instances, mind you this was a complete reboot of both servers due to windows updates...

**Comment by mkobit** [comment]
Ok, I think we figured this out.

1. We previously did not have `https` enabled
1. Webhook URLs were pointing to http://jenkins.internal.com/bitbucket-scmsource-hook/notify
1. We added `https` to our Jenkins instance, and redirected `http` requests to it. Webhooks still pointed to `http`. Things continued to work
1. We upgraded to **1.4.2**
1. Webhooks seemingly stopped working, and no logs were being thrown
1. We changed the webhook URLs to `https`, and it seems to work again
